src/lib/idb.ts

```diff
@@ -165,140 +167,39 @@ export default class IDBStorage {
-    return this.openDatabase().then((db) => {
-      //this.log('save:', entryName, value);
-
-      const handleError = (error: Error) => {
-        this.log.error('save: transaction error:', entryName, value, db, error, error && error.name);
-        if((!error || error.name === 'InvalidStateError')/*  && false */) {
-          setTimeout(() => {
-            this.save(entryName, value);
-          }, 2e3);
-        } else {
-          //console.error('IndexedDB saveFile transaction error:', error, error && error.name);
-        }
-      };
-
-      return new Promise<void>((resolve, reject) => {
-        try {
-          const transaction = db.transaction([this.storeName], 'readwrite');
-  
-          transaction.onerror = (e) => {
-            handleError(transaction.error);
-            reject(transaction.error);
-            clearTimeout(timeout);
-          };
-  
-          transaction.oncomplete = (e) => {
-            this.log('save: transaction complete:', entryName);
-            resolve();
-            clearTimeout(timeout);
-          };
-
-          const timeout = setTimeout(() => {
-            this.log.error('save: transaction not finished', entryName, transaction);
-          }, 10000);
-  
-          /* transaction.addEventListener('abort', (e) => {
-            //handleError();
-            this.log.error('IndexedDB: save transaction abort!', transaction.error);
-          }); */
-  
-          const objectStore = transaction.objectStore(this.storeName);
-
-          if(!Array.isArray(entryName)) {
-            entryName = [].concat(entryName);
-            value = [].concat(value);
-          }
-
-          for(let i = 0, length = entryName.length; i < length; ++i) {
-            const request = objectStore.put(value[i], entryName[i]);
-            request.onerror = (error) => {
-              reject(transaction.error);
-              clearTimeout(timeout);
-            };
-          }
-        } catch(error) {
-          handleError(error);
-          reject(error);
-          
-          /* this.storageIsAvailable = false;
-          throw error; */
-        }
-      });
-    });
+    // const handleError = (error: Error) => {
+    //   this.log.error('save: transaction error:', entryName, value, db, error, error && error.name);
+    //   if((!error || error.name === 'InvalidStateError')/*  && false */) {
+    //     setTimeout(() => {
+    //       this.save(entryName, value);
+    //     }, 2e3);
+    //   } else {
+    //     //console.error('IndexedDB saveFile transaction error:', error, error && error.name);
+    //   }
+    // };
+
+    if(!Array.isArray(entryName)) {
+      entryName = [].concat(entryName);
+      value = [].concat(value);
+    }
+    
+    return this.getObjectStore('readwrite', (objectStore) => {
+      return (entryName as string[]).map((entryName, idx) => objectStore.put(value[idx], entryName));
+    }, DEBUG ? 'save: ' + entryName.join(', ') : '');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MikeOwino/tweb/pullRequest?prid=7329411)
```suggestion
    }, DEBUG ? "save: " + entryName.join(', ') : '');
```